### PR TITLE
[MI-1837] used shopify order number instead of order id on filtering orders

### PR DESCRIPTION
**Pull request by kcasas**
/cc @zendesk/mintegrations

### Description
* Use order number instead of order id when using shopify order id filter
shopify-app-filter-order-id

### References
* JIRA: https://zendesk.atlassian.net/browse/MI-1837

### Risks
* [low] might be slower cause of the extra api call

### CODE REVIEW

File: src/Components/App/index.jsx [hunk 1]
```diff
@@ -40,8 +40,9 @@ class App extends React.Component {
         return (
           <div>
             <Customer {...this.props.customer} />
-            {this.props.orders.length &&
-              <Orders orders={this.props.orders} ordersUri={this.props.ordersUri} />}
+            {this.props.orders.length > 0 &&
```
Comment (kcasas) [inline src/Components/App/index.jsx]: Without `> 0`, if order is empty a number `0` is displayed

File: src/Sagas/orders.js [hunk 2]
```diff
@@ -4,24 +4,26 @@ import { FETCH_ORDER, FETCH_ORDERS, FETCH_ORDERS_FAILED } from '../Actions/const
...
+    data.limit = 3;
   }
 
   yield put(setLoadingState(true));
 
   try {
-    const response = yield call(callApi, endpoint, data);
+    let response = yield call(callApi, endpoint, data);
+    if (action.type === FETCH_ORDER) {
+      const order = response.orders.find(cur => cur.order_number - action.orderId === 0);
```
Comment (kcasas) [inline src/Sagas/orders.js]: Alternative comparison to `cur.order_number === action.orderId`. Problems arise when the other one is a string while the other is a number. Eslint also doesn't want `==` as comparison

Comment (joseconsador) [reply src/Sagas/orders.js]: @kcasas what happens if the order number isn’t part of the response? Example if the shop has thousands of orders, I dont think the Shopify API will return all orders. Should we traverse all pages until the order is found?

**Comment by samgavinio** [comment]
Seems to be missing some of the stuff in the acceptance criteria:

- [] Order ID: example: 4565337033 from the link (https://z3nmgalang.myshopify.com/admin/orders/4565337033)
- [x]Order Name: 1004 in the image
- [] Order Name prefixed with the # symbol: #1004 in the image

The first one in there should work without having to check the recent orders because it doesn't need to.

File: src/Sagas/orders.js (hunk 2 above)
Comment (miogalang) [reply src/Sagas/orders.js]: I think we should just check the first page. We don't want to loop through all of it, that would potentially be very painful and slow. just say not found, and if we get complaints then we think about it

**Pull request merged**